documentation/projects/proposals/project_improvement/20230914-implementation_plan_project_automations.md

```diff
@@ -0,0 +1,262 @@
+that were "classic" GitHub Projects.
+
+- [Openverse](https://github.com/orgs/WordPress/projects/3)
+- [Openverse PRs](https://github.com/orgs/WordPress/projects/8)
+
+```{note}
+These classic project boards have been closed. They are no longer updated, nor
+referred to in any discussions.
+```
+
+These boards were highly automated. A number of automations related to these
+boards were deleted ([PR](https://github.com/WordPress/openverse/pull/2338))
+when they were closed. Those automations can serve as inspiration for their
+reboots for the new project board.
+
+## To-do
+
+The complete work of automating our boards can be broadly classified under three
+categories.
+
+### General improvements to boards
+
+1. Normalise emoji usage
+
+   The backlog board uses an emoji for each column. None of the other boards do.
+   Personally I appreciate them and would like all boards to use them but
+   wouldn't mind dropping them from the backlog board for consistency's sake.
+
+2. Column changes
+
+   - **Issues**
+     - Create new column "🗑️ Discarded" for issues that were closed without
+       resolution (like label "⛔️ status: discarded").
+   - **PRs**
+     - Split "Needs Review" into two columns based on their number of existing
+       reviews.
+       - 2️⃣ Needs 2 Reviews (no reviews yet)
+       - 1️⃣ Needs 1 Review (one approval, near finish line)
+     - Create new column "🗑️ Discarded" for PRs that were closed without merge.
+     - Rename "Merged" to "✅ Done" for consistency with issues board.
+
+### Improvements to existing automations
+
+1. [Issues/Issue is created](/meta/project_boards/issues.md#issue-is-created)
+   and
+   [Issues/Issue is added to the project](/meta/project_boards/issues.md#issue-is-added-to-the-project)
+
+   The issue should be added to the project board under the appropriate column,
+   "📋 Backlog" or "⛔ Blocked" depending on the issue labels.
+
+2. [Issues/Issue is closed](/meta/project_boards/issues.md#issue-is-closed)
+
+   The issue should be moved to the "✅ Done" column only if it was resolved. If
+   it was closed as discarded, it should be moved to a new "🗑️ Discarded"
+   column.
+
+   The built-in workflow will have to be replaced by a custom one that looks at
+   the `state_reason` field inside the issue payload of the `closed` activity of
+   the
+   [`issues` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues).
+
+3. [Issues/Issue is closed and inactive](/meta/project_boards/issues.md#issue-is-closed-and-inactive)
+
+   The current system means that issues are archived one by one as they become
+   inactive and with an 8 day cadence, this will slowly drift out of sync with
+   the weekly development chat.
+
+   Archiving should be performed by a custom scheduled workflow that archives
+   all items in the "✅ Done" and "🗑️ Discarded" columns weekly (preferably
+   immediately after the chat).
```

Comment:
4. Issues/ label changed:
We need to update the `priority` field value if a priority label was added/changed. Currently, when we update the priority of an issue in a label, the custom field value is not updated, so the project columns (or groupings) are not updated accordingly.
The `blocked` label addition/removal should probably also change the `status` custom field.